src/AutoRestNuGetGenerator/AutoRestNuGetGenerator/AutoRestNuGetGenerator.ps1

```diff
@@ -41,11 +41,15 @@ try {
 	$env:path+=";C:\Windows\ServiceProfiles\NetworkService\AppData\Roaming\npm"
 	$env:path+=";$env:SYSTEM_DEFAULTWORKINGDIRECTORY\$packageDirName\tools"
 
-	Write-Output "invoking 'npm install -g autorest@latest'"
-	npm install -g autorest@latest
+	#if autorest is installed, skip the set up
+	$autorestNPMCheck = npm ls -g autorest
+	Write-Host $autorestNPMCheck
 
-	#Write-Output "invoking 'autorest --reset'"
-	#autorest --reset --verbose --debug
+	if (!$autorestNPMCheck -or $autorestNPMCheck.Count -lt 2 -or !$autorestNPMCheck[1].Contains("autorest"))
```

Comment:
Seems to be a lot of checks? but i know it can be awkward the way npm ls returns strings

Reply:
yeah I wanted to be safe... it returns something like
c:\program files\nodejs
`-- autorest@2.0.4283

so checking or null and at least two lines being there ....
better safe than sorry :D

thanks for review